### PR TITLE
Fix(eos_designs): core_interfaces generates invalid config if ASN is not defined for the p2p_links/p2p_links_profiles

### LINKED ISSUE #3596: core_interfaces can generate invalid config if BGP AS is not defined
### Issue Summary

When using core_interfaces, if an ASN is required but not set, eos_designs will generate an invalid config:
Configuration:
```
p2p_links:
    - nodes: [ BA_SPLINE01, WAN_ROUTER ]
      profile: WAN_link
      id: 1
```
Variable `as` is not defined under the profile.
No error is thrown and generated configuration is:
```
   neighbor 172.16.0.9 remote-as None
   neighbor 172.16.0.9 local-as None no-prepend replace-as
 ```

### Which component(s) of AVD impacted

eos_designs

### How do you run AVD ?

Ansible CLI (with virtual-env or native python)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Shivani-gslab**
## Change Summary

Current code is converting the `bgp_as` value to `string`, where `None` value is also getting converted to type `string`, which is revoking the check for None value to raise the `AristaAvdMissingVariableError` even if `core_interfaces.p2p_links.[].as` is not defined.

Fixing the code and tests for the same.

## Related Issue(s)

Fixes #3596 

## Component(s) name

`arista.avd.eos_designs`

## Proposed changes

Fixing the type conversion in `ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py`

## How to test

Test by not defining `core_interfaces.p2p_links.[].as` in input and run the eos_designs role. `AristaAvdMissingVariableError` is expected.

Check the test added in `eos_designs_negative_unit_tests` molecule scenario.

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py [hunk 1]
```diff
@@ -151,8 +151,8 @@ def _get_p2p_data(self, p2p_link: dict) -> dict:
             "peer_type": peer_type,
             "ip": ip[index],
             "peer_ip": ip[peer_index],
-            "bgp_as": str(bgp_as[index]),
-            "peer_bgp_as": str(bgp_as[peer_index]),
+            "bgp_as": (str(bgp_as[index]) if bgp_as[index] is not None else None),
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py]:
```suggestion
            "bgp_as": (str(bgp_as[index]) if bgp_as[index] else None),
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py [hunk 2]
```diff
@@ -151,8 +151,8 @@ def _get_p2p_data(self, p2p_link: dict) -> dict:
...
+            "peer_bgp_as": str(bgp_as[peer_index] if bgp_as[peer_index] is not None else None),
```
Comment (laxmikantchintakindi) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py]:
```suggestion
            "peer_bgp_as": str(bgp_as[peer_index] if bgp_as[peer_index] else None),
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py [hunk 3]
```diff
@@ -151,8 +151,8 @@ def _get_p2p_data(self, p2p_link: dict) -> dict:
             "peer_type": peer_type,
             "ip": ip[index],
             "peer_ip": ip[peer_index],
-            "bgp_as": str(bgp_as[index]),
-            "peer_bgp_as": str(bgp_as[peer_index]),
+            "bgp_as": (str(bgp_as[index]) if bgp_as[index] else None),
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py]: do we really need str here?

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py [hunk 4]
```diff
@@ -32,7 +32,7 @@ def router_bgp(self) -> dict | None:
                 continue
 
             if p2p_link["data"]["bgp_as"] is None or p2p_link["data"]["peer_bgp_as"] is None:
-                raise AristaAvdMissingVariableError("l3_edge.p2p_links.[].as")
+                raise AristaAvdMissingVariableError(f"{self.data_model}.p2p_links.[].as or {self.data_model}.p2p_links_profiles.[].as")
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py:35]: Can we have better error message. Like add index of p2p_links. 
Also if i set only one as then getting below error
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: IndexError: list index out of range
fatal: [core-4-multicast -> localhost]: FAILED! => {"changed": false, "msg": "list index out of range"}
```

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py (hunk 3 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/utils.py]: it is for something in schema conversion

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py (hunk 4 above)
Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py:35]: Fixed the index out of range error.
For error message we can not get the `index` in the current code implementation. So keeping it like this.

**Review by MaheshGSLAB — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py [hunk 5]
```diff
@@ -32,7 +32,7 @@ def router_bgp(self) -> dict | None:
                 continue
 
             if p2p_link["data"]["bgp_as"] is None or p2p_link["data"]["peer_bgp_as"] is None:
-                raise AristaAvdMissingVariableError("l3_edge.p2p_links.[].as")
+                raise AristaAvdMissingVariableError(f"{self.data_model}.p2p_links.[].as or core_interfaces.p2p_links_profiles.[].as")
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py]: Please use the correct data model key in the var.

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_designs/python_modules/core_interfaces_and_l3_edge/router_bgp.py]: Fixed!

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**